File: scratch/analyze_geojson.js
const fs = require('fs');
const path = 'g:\\My Drive\\Web App\\UrbanTree\\urban-tree\\public\\data\\phuong_xa.json';

const data = JSON.parse(fs.readFileSync(path, 'utf8'));
const features = data.features || [];

console.log(`Total features: ${features.length}`);

for (let i = 0; i < Math.min(10, features.length); i++) {
    console.log(`Feature ${i} properties:`, JSON.stringify(features[i].properties));
}

const named = features.find(f => {
    const p = f.properties || {};
    const name = p.name || p.Name || p.ten_phuong || p.description || "";
    return name && !name.includes('PA168');
});

if (named) {
    console.log('Found named feature:', JSON.stringify(named.properties));
}

File: scratch/analyze_geojson.py
import json
import os

path = r'g:\My Drive\Web App\UrbanTree\urban-tree\public\data\phuong_xa.json'
with open(path, 'r', encoding='utf-8') as f:
    data = json.load(f)

features = data.get('features', [])
print(f"Total features: {len(features)}")

# Check first 5 features
for i, f in enumerate(features[:10]):
    props = f.get('properties', {})
    print(f"Feature {i} properties: {props}")

# Find one with a real name
for f in features:
    props = f.get('properties', {})
    name = props.get('name') or props.get('Name') or props.get('ten_phuong')
    if name and 'PA168' not in name:
        print(f"Found named feature: {props}")
        break
